lib/compiler.js

```diff
@@ -149,8 +149,8 @@ function findScripts(dir) {
 
   var results = [];
   files.forEach(function(filename) {
-    // ignore index.js and files prefixed with underscore
-    if ((filename === 'index.js') || (filename[0] === '_')) {
+    // ignore files prefixed with underscore
```

Comment:
When looking for boot scripts in boot directories, the previous code, had ignored ```index.js```. Now that we resolve relative paths of boot directories, before searching for boot scripts, I wonder, if we still must be ignoring ```index.js```. 

If, per the previous code, ```index.js``` is ignored here, then for a scenario like 
```app.boot(app, { appRootDir: __dirname, bootScripts: ["./custom-boot/script" ] })```, where ```script``` is a directory, ```findScripts()``` won't be able to explore ```__dirname+"/custom-boot/script/index.js"```

So, I have modified code here to stop  ignoring ```index.js```.

@bajtos - let me know otherwise.